#82 Deploy extension registry (#88)
exasol/extension-manager: doc/developers_guide.md

Make sure to not commit the modified `package.json`.

## Extension Registry

The extension registry is an HTTPS service that provides a JSON file containing links to all available extensions. It consists of an S3 Bucket and a CloudFront distribution deployed via CDK.

### Initial Configuration

1. Create file `registry/lib/config.ts` with the following content:
    ```ts
    export const CONFIG = {
        owner: 'your.email@example.com'
    }
    ```
2. Run `npm install`
3. Configure AWS profile and region:
    ```sh
    export AWS_PROFILE=<profile>
    export AWS_REGION=eu-central-1
    ```

### Deploy Changes

Run `npm run cdk diff`. If the output looks good, run `npm run cdk deploy`.

To get the output variables of the deployed stack (e.g. bucket name and CloudFront distribution host name), run the following command:

```sh
aws cloudformation describe-stacks --stack-name ExtensionManagerRegistry --query "Stacks[0].Outputs[].{key:ExportName,value:OutputValue}"
```

### Deploy Registry Content

> To deploy the content of the Extension Registry to `test` or `prod` stage, run script `AWS_PROFILE=<profile> upload-registry.sh test|prod`. This will upload one of the json files in the `content` folder to the S3 bucket and invalidate the CloudFront cache.

Reviewer: ```suggestion
To deploy the content of the Extension Registry to `test` or `prod` stage, run script `AWS_PROFILE=<profile> upload-registry.sh <prefix>` with `<prefix>` being either `test` or `prod`. This will upload the json file with the specified prefix into the `content` folder to the S3 bucket and invalidate the CloudFront cache.
```

Reviewer: We could also call the variable `<stage>` and explain that this variable is also interpreted as a prefix of the respective file to upload.
